### PR TITLE
Provide content assists for sharing URLs

### PR DESCRIPTION
**Pull request by yschimke**
See https://twitter.com/yrezgui/status/1416673883437113350

### CODE REVIEW

**Comment by yschimke** [comment]
Relevant links

https://developers.google.com/assistant/app/read-it
https://developer.android.com/reference/android/app/Activity#onProvideAssistContent(android.app.assist.AssistContent)
https://developer.android.com/training/articles/assistant

**Comment by yschimke** [comment]
image

File: app/src/main/java/com/gh4a/activities/CommitActivity.java [hunk 1]
```diff
@@ -237,4 +237,15 @@ private void loadComments(boolean force) {
                     showContentIfReady();
                 }, this::handleLoadFailure);
     }
+
+    @Nullable
+    @Override
+    protected Uri getActivityUri() {
+        if (mPullRequestNumber > 0)
+        return Uri.parse("https://github.com/" + mRepoOwner + "/" + mRepoName + "pull/" +
```
Comment (maniac103) [inline app/src/main/java/com/gh4a/activities/CommitActivity.java:245]: Suggestion: Start with `IntentUtils.createBaseUriForRepo()` and `append()` to that. See `onOptionsItemSelected` for an example. Doing so also allows sharing most of the generation in the two branches:
```
Uri.Builder builder = IntentUtils.createBaseUriForRepo(mRepoOwner, mRepoName);
if (mPullRequestNumber > 0) {
    builder.appendPath("pull")
            .appendPath(String.valueOf(mPullRequestNumber));
}
return builder.appendPath("commit")
        .appendPath(mObjectSha)
        .build();
```

File: app/src/main/java/com/gh4a/activities/CommitActivity.java [hunk 2]
```diff
@@ -237,4 +237,15 @@ private void loadComments(boolean force) {
...
+                mPullRequestNumber + "/commit/" + mObjectSha);
+        else
```
Comment (maniac103) [inline app/src/main/java/com/gh4a/activities/CommitActivity.java:247]: Please add curly braces in all if-else cases.

File: app/src/main/java/com/gh4a/activities/CommitActivity.java (hunk 1 above)
Comment (yschimke) [reply app/src/main/java/com/gh4a/activities/CommitActivity.java]: Slight spin on what you've suggested but the base uri methods and Uri.Builder approach look better.

**Comment by yschimke** [comment]
I haven't retested since addressing review comments.

File: app/src/main/java/com/gh4a/activities/CommitActivity.java (hunk 1 above)
Comment (maniac103) [reply app/src/main/java/com/gh4a/activities/CommitActivity.java]: Why return the `Uri.Builder` though? The activity doesn't intend to modify the URI after all.

File: app/src/main/java/com/gh4a/activities/CommitActivity.java [hunk 3]
```diff
@@ -237,4 +239,14 @@ private void loadComments(boolean force) {
...
     }
+
+    @Nullable
+    @Override
+    protected Uri.Builder getActivityUri() {
+        Uri.Builder builder = createBaseUriForRepo(mRepoOwner, mRepoName);
+        if (mPullRequestNumber > 0) {
+            builder.appendPath("pull").appendPath(String.valueOf(mPullRequestNumber));
+        }
+        return builder.appendPath("commit").appendPath(mObjectSha);
```
Comment (maniac103) [inline app/src/main/java/com/gh4a/activities/CommitActivity.java:250]: Style nit: please put the chained calls on new lines

File: app/src/main/java/com/gh4a/activities/CommitActivity.java [hunk 4]
```diff
@@ -41,6 +41,8 @@
 
 import java.util.List;
 
+import static com.gh4a.utils.IntentUtils.createBaseUriForRepo;
```
Comment (maniac103) [inline app/src/main/java/com/gh4a/activities/CommitActivity.java:44]: Style not: please avoid static imports.

**Comment by maniac103** [comment]
Thanks!

**Comment by yschimke** [comment]
@maniac103 I know share is there, but 50% of the time I go to share then navigate to the Copy URL option.  This makes the app switcher much more useful as well as screenshots, and copying text out of images, it also has a URL to copy. See whether you find this useful also.

**Pull request merged**